File: backend/internal/model/category.go
package model

import "time"

// Category represents a transaction category (income or expense).
type Category struct {
	ID        uint64    `db:"id" json:"id"`
	UserID    uint64    `db:"user_id" json:"user_id"`
	Name      string    `db:"name" json:"name"`
	Type      string    `db:"type" json:"type"` // income | expense
	IsSystem  int8      `db:"is_system" json:"is_system"`
	IsActive  int8      `db:"is_active" json:"is_active"`
	CreatedAt time.Time `db:"created_at" json:"created_at"`
	UpdatedAt time.Time `db:"updated_at" json:"updated_at"`
}

const (
	CategoryTypeIncome  = "income"
	CategoryTypeExpense = "expense"
)

// DefaultExpenseCategories are pre-created for new users.
var DefaultExpenseCategories = []string{
	"餐饮", "交通", "购物", "住房", "娱乐", "医疗", "教育", "通讯", "其他支出",
}

// DefaultIncomeCategories are pre-created for new users.
var DefaultIncomeCategories = []string{
	"工资", "奖金", "投资收益", "兼职", "其他收入",
}

File: frontend/lib/models/category.dart
class Category {
  final int id;
  final int userId;
  final String name;
  final String type;
  final int isSystem;
  final int isActive;
  final String createdAt;
  final String updatedAt;

  Category({
    required this.id,
    required this.userId,
    required this.name,
    required this.type,
    required this.isSystem,
    required this.isActive,
    required this.createdAt,
    required this.updatedAt,
  });

  bool get system => isSystem == 1;
  bool get active => isActive == 1;
  bool get isIncome => type == 'income';
  bool get isExpense => type == 'expense';

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'] as int,
      userId: json['user_id'] as int,
      name: json['name'] as String,
      type: json['type'] as String,
      isSystem: json['is_system'] as int,
      isActive: json['is_active'] as int,
      createdAt: json['created_at'] as String,
      updatedAt: json['updated_at'] as String,
    );
  }
}

